### PR TITLE
Add Reducer

### PR DESCRIPTION
**Pull request by danutibe07**
- Use store, actions, and reducers in React.
- Use Duck Pattern for redux files

### CODE REVIEW

File: src/redux/books/BookSlice.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const initialState = {
+  books: [],
+};
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, { payload }) => {
+      state.books.push({ title: payload.title, author: payload.author });
```
Comment (omarsalem7) [inline src/redux/books/BookSlice.js]: - [ ] Kindly consider adding an `id` property, so when you try to remove the book you can select it by the id but in your case, the book doesn't have any book.  You can use a random method like this `  id: Math.floor(Math.random() * 100),`

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @danutibe07 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️

- [ ] _Check the comments under the review._

- [ ] Kindly consider fixing the broken [MIT license](https://github.com/danutibe07/React-BookStore/tree/addredux#:~:text=This%20project%20is%20MIT%20licensed) in the readme to make it professional

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
...

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @danutibe07 👍, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Use store, actions, and reducers in React ✔️ 
Structure application files using the ducks pattern for your Redux files ✔️ 

 ## Required Changes ♻️

**- [ ] Please kindly fix the [MIT license](https://choosealicense.com/licenses/mit/) broken link in your README file as pointed out earlier. This will make your repo informative.**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @danutibe07, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewers. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## To Highlight 👍

- Has a proper Pull Request. ✔️
- Action and Reducer functionality. ✔️
- Ducks pattern file structure. ✔️
- @reduxjs/toolkit & react-redux present in package.json ✔️
- Linters environment properly setup ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**